Core/automation/lib/python/core/date.py

```diff
@@ -269,6 +274,9 @@ def to_java_zoneddatetime(value):
     # Eclipse Smarthome DateTimeType
     if EclipseDateTimeType and isinstance(value, EclipseDateTimeType):
         return to_java_zoneddatetime(value.calendar)
+    # openHAB DateTimeItem
+    if DateTimeItem and isinstance(value, DateTimeItem):
+        return to_java_zoneddatetime(value.getState())
```

Comment:
Lets move this to the end of the list of conversions and add another to catch Eclipse Items as well.